### DISCUSSION TITLE
Font Size

### DISCUSSION THREAD

**Discussion by tomaculum** [Spirik/GEM#70] [Q&A] [answered]
Hi all,
I am using the Adafruit GFX part of this library. As the documentation or some example states, `menu.init();` sets the font size to `1`. Unfortunately this is very small on my display. Is there any way to increase the font size?
Best regards
Tom

**Comment by Spirik** [comment, marked as answer]
Hi Tom!

Unfortunately, currently there is no easy way to change font scaling w/o editing source code of GEM. I really should add this feature in future updates (although there is one caveat that I have to figure something out about, see below). That being said, here is small rundown of the changes in source code that can be made to scale font size 2x times:
* Change font dimensions here (multiply each dimension by 2, except for the last one in each triplet: it can be multiplied by 2 with additional adjustments for better vertical alignment): https://github.com/Spirik/GEM/blob/master/src/GEM_adafruit_gfx.h#L153 ;
* Change call to `_agfx.setTextSize(2)` here: https://github.com/Spirik/GEM/blob/master/src/GEM_adafruit_gfx.cpp#L153 and here: https://github.com/Spirik/GEM/blob/master/src/GEM_adafruit_gfx.cpp#L191 ;
* Change offset calculations here (multiply each number by 4): https://github.com/Spirik/GEM/blob/master/src/GEM_adafruit_gfx.cpp#L251 ;
* Change call to `_agfx.drawChar()` (replace `1` with `2`) here: https://github.com/Spirik/GEM/blob/master/src/GEM_adafruit_gfx.cpp#L709 .

And in your sketch set [menuItemHeight](https://github.com/Spirik/GEM/wiki/Menu-appearance) appropriately to accommodate enlarged text (e.g. `GEM_adafruit_gfx menu(tft, GEM_POINTER_ROW, GEM_ITEMS_COUNT_AUTO, 24, 24);`).

The limitation of this approach is that it does nothing to enlarge sprites (i.e. pictograms that are used to draw arrows and checkboxes), so the icons will look rather small compared to text. And that is the caveat that I was referring to earlier. I am a little hesitate to add new set of sprites with 2x dimensions risking to further bloat the library size. So I will have to figure something out=)

Here are gists with all necessary edits, you can replace existing `GEM_adafruit_gfx.h` and `GEM_adafruit_gfx.cpp` files bundled with the GEM library with the following ones:
* [GEM_adafruit_gfx.h](https://gist.github.com/Spirik/4d3c7b096b9dc185e3e5c36c07a9e537)
* [GEM_adafruit_gfx.cpp](https://gist.github.com/Spirik/50b2719e75f74632fdaffae511563864)

And to wrap this up, here are some photos.

Before scaling:
![GEM-agfx-text-size_before](https://github.com/Spirik/GEM/assets/1061406/842e0b3b-9eed-4504-a96e-6c856060e18f)

After scaling:
![GEM-agfx-text-size_after](https://github.com/Spirik/GEM/assets/1061406/c98ef1b5-b662-4ddb-9662-c2161cf610fa)

Hope that will help=)

**Reply by tomaculum** [reply]
Thank you very much! This worked great (: 

Previous size:
![IMG_1537 Kopie](https://github.com/Spirik/GEM/assets/16617007/cdbef2b7-e3bc-4929-920a-15c1e5eacd1d)

After scaling:
![IMG_1539 Kopie](https://github.com/Spirik/GEM/assets/16617007/9724e288-cb60-406f-bd40-58de00552882)

**Reply by Spirik** [reply]
Glad to help! I will add this feature to release version of GEM eventually=)

**Comment by Spirik** [comment]
Hi, @tomaculum !

Not sure if still relevant, but just want to let you know that setting text size for Adafruit GFX version of GEM is now available through `menu.setTextSize()` method since version [1.4.2](https://github.com/Spirik/GEM/releases/tag/1.4.2), see [wiki](https://github.com/Spirik/GEM/wiki/Menu-appearance#changing-text-magnification-adafruit-gfx-version-only) for additional details.

Have a good day!

**Reply by tomaculum** [reply]
Great, thank you very much! Will try it in the upcoming days (:
